src/kixi/heimdall/components/metrics.clj

```diff
@@ -11,6 +11,61 @@
             [metrics.ring.expose :as expose]
             [taoensso.timbre :as log]))
 
+(def upper-name
+  (memoize
+    (fn [level]
+      (str/upper-case (name level)))))
+
+(def uri-method-status->metric-name
+  (memoize
+    (fn [uri method status]
+      (let [name (-> (str "." (upper-name method))
+                     safe-name
+                     (str "." status))]
+        ["resources" (safe-name uri) name]))))
+
+(defn request-ctx->metric-name
+  [ctx]
+  (uri-method-status->metric-name
+    (get-in ctx [:request :uri])
+    (:method ctx)
+    (get-in ctx [:response :status])))
```

Comment:
are we OK if any of these are nil? Do we want a different default?

Reply:
Not sure what default though:  this is lifted from datastore code that's been working happily for a good whilst.  Happy to leave as is.